File: openmp/share3.f90
program workshare_example
! use omp_lib
  implicit none
  integer, parameter :: N = 10
  integer :: i
  real :: array(N), result(N)
  do i = 1, N
    array(i) = real(i)
  end do

  !$omp parallel shared(array, result)
  !$omp workshare
  result = array * array
  !$omp end workshare
  !$omp end parallel

  print *, 'Squared elements of array:'
  do i = 1, N
    print *, result(i)
  end do
end program workshare_example

File: openmp/share3.c
// share3.c
#include <stdio.h>
#include <omp.h>

#define N 10

int main() {
  int i, array[N], result[N];
  for (i = 0; i < N; i++) {
    array[i] = i + 1;
  }
  #pragma omp parallel shared(array, result) private(i) 
  {
    int tid = omp_get_thread_num();
    #pragma omp for
    for (i = 3; i < N; i++) {
      printf("tid=%d i=%d\n", tid, i);
      result[i] = array[i] * array[i];
      //if (i > 5) break;
    }
  }

  printf("Squared elements of array:\n");
  for (i = 0; i < N; i++) {
    printf("%d ", result[i]);
  }
  printf("\n");
  return 0;
}

